crates/bevy_text/src/glyph_brush.rs

```diff
@@ -0,0 +1,178 @@
+                    min_x = min_x.min(glyph.position.x - scaled_font.h_side_bearing(glyph.id));
+                }
+                max_y = max_y.floor();
+                min_x = min_x.floor();
+
+                for section_glyph in section_glyphs {
+                    let handle = &self.handles[section_glyph.font_id.0];
+                    let font = fonts.get(handle).ok_or(TextError::NoSuchFont)?;
+                    let glyph_id = section_glyph.glyph.id;
+                    let font_size = section_glyph.glyph.scale.y;
+                    if let Some(outlined_glyph) = font.font.outline_glyph(section_glyph.glyph) {
+                        let bounds = outlined_glyph.px_bounds();
+                        let handle_font_atlas: Handle<FontAtlasSet> = handle.as_weak();
+                        let font_atlas_set = font_atlas_set_storage
+                            .get_or_insert_with(handle_font_atlas, FontAtlasSet::default);
+
+                        let atlas_info = font_atlas_set
+                            .get_glyph_atlas_info(font_size, glyph_id)
+                            .map(Ok)
+                            .unwrap_or_else(|| {
+                                font_atlas_set.add_glyph_to_atlas(
+                                    texture_atlases,
+                                    textures,
+                                    outlined_glyph,
+                                )
+                            })?;
+
+                        let texture_atlas = texture_atlases.get(&atlas_info.texture_atlas).unwrap();
+                        let glyph_rect = texture_atlas.textures[atlas_info.glyph_index as usize];
+                        let glyph_width = glyph_rect.width();
+                        let glyph_height = glyph_rect.height();
+
+                        let x = bounds.min.x + glyph_width / 2.0 - min_x;
+                        // the 0.5 accounts for odd-numbered heights (bump up by 1 pixel)
+                        // max_y = text block height, and up is negative (whereas for transform, up is positive)
+                        let y = max_y - bounds.max.y + glyph_height / 2.0 + 0.5;
+                        let position = Vec2::new(x, y);
+
+                        vertices.push(TextVertex {
+                            position,
+                            atlas_info,
+                        });
+                    }
+                }
+                Ok(vertices)
+            })
+            .collect::<Result<Vec<_>, TextError>>()?
+            .into_iter()
+            .flatten()
+            .collect();
+        Ok(vertices)
+    }
+
+    pub fn add_font(&mut self, handle: Handle<Font>, font: FontArc) -> FontId {
+        self.fonts.push(font);
+        self.handles.push(handle);
+        let font_id = self.latest_font_id;
+        self.latest_font_id = FontId(font_id.0 + 1);
+        font_id
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct TextVertex {
```

Comment:
It seems like "vertex" isn't the right name for this. Vertices are points that make up a mesh, but this seems more like a `PositionedGlyph`.